### ISSUE TITLE
Question about straight-through estimator implementation and trainable modules during InstructPix2Pix fine-tuning

### ISSUE THREAD

**Issue by Terry01001** [Shilin-LU/VINE#19]
Hi Shilin,
Thank you for releasing the code for VINE.

I am trying to understand the third-stage fine-tuning procedure for VINE-R described in the paper, where InstructPix2Pix is incorporated into the noise layer and gradients are backpropagated using a straight-through estimator (STE).

I looked through the released code, especially `vine/src/finetune.py`, and I have two questions:

1. Could you point me to where the STE is implemented in the released code?

   In `finetune.py`, I found the InstructPix2Pix pipeline being used around the fine-tuning loop, but I could not find an explicit STE-style operation such as:

   ```python
   transformed = encoded + (edited - encoded).detach()
   ```

   or an equivalent custom autograd implementation. I may have missed it, so I would appreciate it if you could clarify where this part is implemented.

2. During the InstructPix2Pix fine-tuning stage, are both the watermark encoder and decoder updated, or is only the decoder fine-tuned?

   From my reading of `finetune.py`, it seems that `watermark_encoder.sec_encoder`, `watermark_encoder.unet`, and `watermark_encoder.vae_a2b` are set to `requires_grad_(False)`, while the optimizer is constructed only from `decoder.parameters()`. Does this mean the released fine-tuning script only updates the decoder?

I am also curious about the training strategy when using STE with image editing models in the noise layer. If the encoder is trained through STE, do you recommend freezing some parts of the encoder, such as parts of the UNet/VAE/condition adaptor, or did you find that fine-tuning only the decoder is more stable or effective in practice?

Any insight would be very helpful. Thanks again for the great work.
